src/RecordManager/Finna/Splitter/Ead3.php

```diff
@@ -96,29 +96,33 @@ public function __construct($params)
     public function setData($data)
     {
         $this->doc = \RecordManager\Base\Utils\MetadataUtils::loadXML($data);
+
         $this->recordNodes = $this->doc->xpath('archdesc | archdesc/dsc//*[@level]');
         $this->recordCount = count($this->recordNodes);
         $this->currentPos = 0;
 
         $this->agency
             = (string)$this->doc->control->maintenanceagency->agencycode;
-        $this->archiveId = urlencode(
-            (string)($this->doc->control->recordid
-                ? $this->doc->control->recordid
-                : $this->doc->control->recordid)
-        );
-
-        $this->archiveTitle
-            = (string)$this->doc->control->filedesc->titlestmt->titleproper;
 
-        if (!$this->archiveTitle) {
-            $this->archiveTitle = $this->archiveId;
+        foreach ($this->doc->archdesc->did->unitid as $i) {
```

Comment:
Onko tarkoituksella niin, ettei $this->archiveId voi enää tulla $this->doc->control->recordid:stä?

Reply:
Kyllä, koska AHAAn uudessa siirtoformaatissa $this->doc->control->recordid ei vastaa arkiston unitId:tä. Vaikka aiempi versio toimisi joillain tietolähteillä, ajattelin että on yksinkertaisempi hakea aina samasta paikasta.